core/src/main/java/com/sequenceiq/cloudbreak/job/stackpatcher/ExistingStackPatcherJob.java

```diff
@@ -70,23 +70,26 @@ protected void executeTracedJob(JobExecutionContext context) throws JobExecution
             StackPatchType stackPatchType = stackPatchTypeConverter.convertToEntityAttribute(stackPatchTypeName);
             if (stackPatchType == null || StackPatchType.UNKNOWN.equals(stackPatchType)) {
                 String message = String.format("Stack patch type %s is unknown", stackPatchTypeName);
-                unscheduleAndFail(message, context, stack, stackPatchType);
+                unscheduleAndFailJob(message, context, stack, stackPatchType);
             } else {
                 Optional<ExistingStackPatchService> optionalExistingStackPatchService = getStackPatchServiceForType(stackPatchType);
                 if (optionalExistingStackPatchService.isEmpty()) {
                     String message = "No stack patcher implementation found for type " + stackPatchType;
-                    unscheduleAndFail(message, context, stack, stackPatchType);
+                    unscheduleAndFailJob(message, context, stack, stackPatchType);
                 } else {
-                    applyStackPatch(optionalExistingStackPatchService.get(), stack);
+                    boolean success = applyStackPatch(optionalExistingStackPatchService.get(), stack);
+                    if (success) {
+                        unscheduleJob(context);
+                    }
                 }
             }
         } else {
             LOGGER.debug("Existing stack patching will be unscheduled, because stack {} status is {}", stack.getResourceCrn(), stackStatus);
+            unscheduleJob(context);
         }
-        unscheduleJob(context);
     }
 
-    private void unscheduleAndFail(String message, JobExecutionContext context, Stack stack, StackPatchType stackPatchType)
+    private void unscheduleAndFailJob(String message, JobExecutionContext context, Stack stack, StackPatchType stackPatchType)
```

Comment:
this should have some logs also

Reply:
The message is logged by quartz from the JobExecutionException

Reply:
it would be helpful to see what's the progress, like you log here, and also that is logged too, so if something happens between we know the flow of code for sure

Reply:
Added log message